ports/esp8266/esp_mphal.c

```diff
@@ -128,21 +128,17 @@ void mp_hal_debug_tx_strn_cooked(void *env, const char *str, uint32_t len) {
     }
 }
 
-uint32_t mp_hal_ticks_ms(void) {
+uint32_t MP_FASTCODE(mp_hal_ticks_ms)(void) {
     // Compute milliseconds from 64-bit microsecond counter
     system_time_update();
     return ((uint64_t)system_time_high_word << 32 | (uint64_t)system_time_low_word) / 1000;
 }
 
-uint32_t mp_hal_ticks_us(void) {
-    return system_get_time();
-}
-
-void mp_hal_delay_ms(uint32_t delay) {
+void MP_FASTCODE(mp_hal_delay_ms)(uint32_t delay) {
     mp_hal_delay_us(delay * 1000);
 }
 
-void ets_event_poll(void) {
+void MP_FASTCODE(ets_event_poll)(void) {
```

Comment:
not sure this is worth having in iram given that it calls two functions, one of which is in flash?

Reply:
ah, good point, indeed `mp_handle_pending()` is in iROM and it may call into the VM/runtime so this whole chain could never fit in iRAM.

but note that `ets_loop_iter()` is in iRAM because everything in the `ets_alt_task`c` file is in iRAM (by default)

Reply:
fixed